src/lib.rs

```diff
@@ -225,6 +299,71 @@ impl Address {
     ) -> Address {
         chain_addr::Address(discrimination.into(), chain_addr::Kind::Account(key.0)).into()
     }
+
+    pub fn multisig_from_merkle_root(
+        merkle_root: &[u8],
+        discrimination: AddressDiscrimination,
+    ) -> Result<Address, JsValue> {
+        match merkle_root.len() {
+            32 => {
+                let mut sized_root = [0; 32];
+                sized_root.copy_from_slice(&merkle_root);
+                Ok(chain_addr::Address(
+                    discrimination.into(),
+                    chain_addr::Kind::Multisig(sized_root),
+                )
+                .into())
+            }
+            _ => Err(JsValue::from_str("Invalid merkle root size")),
+        }
+    }
+
+    pub fn get_discrimination(&self) -> AddressDiscrimination {
+        AddressDiscrimination::from(self.0.discrimination())
+    }
+
+    pub fn get_kind(&self) -> AddressKind {
+        AddressKind::from(self.0.to_kind_type())
+    }
+
+    pub fn to_single_address(&self) -> Option<SingleAddress> {
```

Comment:
Since all of these return `Option` I'm open to adding `try_` to the function name if people prefer that.